test/libyul/yulOptimizerTests/fullInliner/no_inline_leave.yul

```diff
@@ -9,8 +9,8 @@
 //
 // {
 //     {
-//         let a_2 := calldataload(0)
-//         a_2 := g()
+//         let a_1 := calldataload(0)
```

Comment:
Need to figure out why this is happening (it's somewhat annoying having to pin another 30+ tests to shanghai), but I'm assuming it's due to the expression splitter (runs before full inliner) not yielding
```
let a_1 := 0
let a_2 := calldataload(a_1)
```
as before, and then having the expression joiner (runs after full inliner) produce:
```
let a_2 := calldataload(0)
```
we're now just getting (as we should):
```
let a_1 := calldataload(0)
```
which is then ignored by the expression joiner, since there's nothing to join. 
I might be mistaken though, and will investigate further before I pin any more tests to shanghai.

Reply:
Yup, just confirmed, that's exactly what happens; in the pre PUSH0 branch we get
```
let _1 := 0                                // expression splitter
let a1 := calldataload(_1)      // expression splitter
let a_2 := a1;                           // full inliner
a_2 := g();                                // full inliner
```
whereas with PUSH0 optimization the expression splitter doesn't do anything since obviously it shouldn't split the `calldataoad` with the zero literal parameter, and the postfix notation for temp variables goes from `_1`, `_2`, ... `_n`, so we end up with the full inliner actually introducing `a_1`.